### PR TITLE
TASK: Make it possible to use DELETE with the curl utility

### PR DESCRIPTION
**Pull request by kabarakh**

### CODE REVIEW

**Comment by kabarakh** [comment]
Wir brauchen den für unser aktuelles Projekt

File: Classes/Utility/Curl/Request.php [hunk 1]
```diff
@@ -79,6 +79,22 @@ public function get()
         return $this->executeRequest($request);
     }
 
+    /**
+     * delete data from a defined URL
+     *
+     * @return Response
+     */
+    public function delete()
```
Comment (daniellienert) [inline Classes/Utility/Curl/Request.php]: Please annotate the return value here

Comment (kabarakh) [reply Classes/Utility/Curl/Request.php]: Done, and went over the whole file to correct return types and the header comment

File: Classes/Utility/Curl/Request.php [hunk 2]
```diff
@@ -125,16 +122,16 @@ public function setUrl($url)
     /**
      * @return string
      */
-    public function getUrl()
+    public function getUrl(): Request
```
Comment (daniellienert) [inline Classes/Utility/Curl/Request.php]: Hier kommt ein String zurück

Comment (kabarakh) [reply Classes/Utility/Curl/Request.php]: du hast vollkommen recht

File: Classes/Utility/Curl/Request.php [hunk 3]
```diff
@@ -55,7 +37,7 @@ class Request
      * @param  string $data
      * @return Response
      */
-    public function post($data = '')
+    public function post($data = ''): Response
```
Comment (andrehoffmann30) [inline Classes/Utility/Curl/Request.php]: string $data, if possible

File: Classes/Utility/Curl/Request.php [hunk 4]
```diff
@@ -101,22 +99,21 @@ protected function buildRequest()
      * @param $request
      * @return Response
      */
-    protected function executeRequest($request)
+    protected function executeRequest($request): Response
```
Comment (andrehoffmann30) [inline Classes/Utility/Curl/Request.php]: has $request no specific type like string or class Request?

File: Classes/Utility/Curl/Request.php [hunk 5]
```diff
@@ -143,9 +140,9 @@ public function setProxy($proxyUrl)
 
     /**
      * @param $verifySSL
-     * @return $this
+     * @return Request
      */
-    public function setVerifySSL($verifySSL)
+    public function setVerifySSL($verifySSL): Request
```
Comment (andrehoffmann30) [inline Classes/Utility/Curl/Request.php]: has $verifySSL no specific type?

**Pull request merged**